Graphics/Slicer/Math/PGA.hs

```diff
@@ -151,27 +220,22 @@ combineConsecutiveLineSegs lines
         sameLineSeg = meet2PLine2 (eToPLine2 l1) (eToPLine2 l2) == PPoint2 (GVec [])
         sameMiddlePoint = p2 == addPoints p1 s1
 
--- | Given the result of intersectionPoint, find out whether this intersection point is on the given segment, or not.
-onSegment :: LineSeg -> Point2 -> Bool
-onSegment (LineSeg p s) i =
-  sqNormOfPLine2 (join2PPoint2 (eToPPoint2 p) (eToPPoint2 i))               <= segmentLength &&
-  sqNormOfPLine2 (join2PPoint2 (eToPPoint2 i) (eToPPoint2 (addPoints p s))) <= segmentLength
+-- | find a point a given distance along a line perpendicularly bisecting this line at a given point.
+pointOnPerp :: LineSeg -> Point2 -> ℝ -> Point2
```

Comment:
This type should be `LineSeg -> R -> Point2 -> Point2` so it forms an endomorphism and can be easily `fmap`ed and `modify`ed.